compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java

```diff
@@ -2057,7 +2059,7 @@ private BRecordTypeSymbol createRecordTypeSymbol(PackageID pkgID, Location locat
         SymbolEnv env = data.env;
         Name genName = Names.fromString(anonymousModelHelper.getNextAnonymousTypeKey(pkgID,
                 data.anonTypeNameSuffixes));
-        BRecordTypeSymbol recordSymbol = Symbols.createRecordSymbol(data.constantSymbol.flags, genName,
+        BRecordTypeSymbol recordSymbol = Symbols.createRecordSymbol(Flags.PUBLIC, genName,
```

Comment:
Alternatively, we can add the public flag to the flag set and set that.

Reply:
This change was done to maintain the consistency between the tuple and record types. Not because the tests were failing. Decided to keep this as it is. Since changing this is effectively removing some other flags present.